### PR TITLE
Built in Async Heal Pool Animations

### PR DESCRIPTION
**Pull request by JT122406**
Built in Async HealPool Animations when someone buys heal pool upgrade.
Is disabled by default but has config option to enable, and config option to allow all teams to see particles or just allow team that bought upgrade to see.  Uses the new #465 to remove particles and save resources when a team is eliminated.

### CODE REVIEW

**Comment by JT122406** [comment]
I have been testing extensively and it works great!

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/BedWars.java [hunk 1]
```diff
@@ -297,6 +298,12 @@ public void onEnable() {
         // Register events
         registerEvents(new EnderPearlLanded(), new QuitAndTeleportListener(), new BreakPlace(), new DamageDeathMove(), new Inventory(), new Interact(), new RefreshGUI(), new HungerWeatherSpawn(), new CmdProcess(),
                 new FireballListener(), new EggBridge(), new SpectatorListeners(), new BaseListener(), new TargetListener(), new LangListener(), new Warnings(this), new ChatAFK());
+
+        Bukkit.getScheduler().runTaskAsynchronously(this, () -> {
```
Comment (andrei1058) [inline bedwars-plugin/src/main/java/com/andrei1058/bedwars/BedWars.java]: what the actual duck? what are you doing here man?

Comment (andrei1058) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/BedWars.java]: this is not ok. why are you doing it in a new thread?

File: bedwars-api/src/main/java/com/andrei1058/bedwars/api/server/VersionSupport.java [hunk 2]
```diff
@@ -435,4 +435,6 @@ public void spigotHidePlayer(Player victim, Player receiver) {
     public abstract void playRedStoneDot(Player player);
 
     public abstract void clearArrowsFromPlayerBody(Player player);
+
+    public abstract void playEffect(Player player, Location location);
```
Comment (andrei1058) [inline bedwars-api/src/main/java/com/andrei1058/bedwars/api/server/VersionSupport.java]: This must be called playVillagerEffect or something like this

**Review by andrei1058 — CHANGES_REQUESTED** [review]

File: bedwars-plugin/src/main/java/com/andrei1058/bedwars/BedWars.java (hunk 1 above)
Comment (JT122406) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/BedWars.java]: I can handle making this change along with the other

Comment (JT122406) [reply bedwars-plugin/src/main/java/com/andrei1058/bedwars/BedWars.java]: That async was actually an accident didn’t mean to commit it I was messing with things

File: bedwars-api/src/main/java/com/andrei1058/bedwars/api/server/VersionSupport.java (hunk 2 above)
Comment (JT122406) [reply bedwars-api/src/main/java/com/andrei1058/bedwars/api/server/VersionSupport.java]: Fixed

**Pull request merged**